### PR TITLE
Wrap DnsResponseException in DnsException

### PR DESCRIPTION
**Pull request by maximcus**
### Description:

Found out that there could be a `DnsResponseException` thrown. Now I wrap it in `DnsException`

### CODE REVIEW

File: Microsoft.Exchange.WebServices.Tests/WithDnsClient.cs [hunk 1]
```diff
@@ -69,6 +69,29 @@ public void SrvQuery_Should_Execute_Srv_Query_For_Specified_Domain()
...
+        dnsQueryClientMock
+            .Setup(dnsQueryClient => dnsQueryClient.Query(srvDomain, QueryType.SRV, QueryClass.IN))
+            .Throws(new DnsResponseException(errorMessage));
+
+        // Act
+        void InvokeSrvQuery() => dnsClient.SrvQuery(domain: "any.domain");
+
+        // Assert
+        var dnsException = Assert.Throws<DnsException>(InvokeSrvQuery);
+        Assert.AreEqual(expected: errorMessage, actual: dnsException!.Message);
```
Comment (adil-mammadov) [inline Microsoft.Exchange.WebServices.Tests/WithDnsClient.cs]: Shouldn't we check inner exception as well?

**Review by adil-mammadov — COMMENTED** [review]
Looks good, I've added one minor suggestion

File: Microsoft.Exchange.WebServices.Tests/WithDnsClient.cs (hunk 1 above)
Comment (maximcus) [reply Microsoft.Exchange.WebServices.Tests/WithDnsClient.cs]: sure

**Review by adil-mammadov — APPROVED** [review]
👍🏼

**Review by pawelmajda — APPROVED** [review]
Looks good

**Pull request merged**